source_gen/lib/src/constants/revive.dart

```diff
@@ -19,8 +19,8 @@ import '../utils.dart';
 /// Dart source code (such as referencing private constructors). It is up to the
 /// build tool(s) using this library to surface error messages to the user.
 Revivable reviveInstance(DartObject object, [LibraryElement origin]) {
-  origin ??= object.type.element.library;
-  final element = object.type.element;
+  final element = object.type.element ?? object.toFunctionValue();
```

Comment:
what version of `analyzer` do we need to ensure that `DartObject` has a `toFunctionValue()` method? it looks like it's missing on travis

Reply:
It was added in https://github.com/dart-lang/sdk/commit/4c08c5026d29784099019b68a2f4e6e655d63ab3, so looks like `0.37.1` (which has been published since then).

Reply:
Should I move the lower bound for the `analyzer` dependency version range in the same PR, or do you prefer a separate one?

Reply:
Same PR is fine.

Reply:
Travis is still not happy :-(

Reply:
Looks like https://github.com/dart-lang/source_gen/blob/master/.travis.yml needs to drop support for 2.1.0.

https://github.com/dart-lang/sdk/commit/bb12829571f064174fab0107851a067e0a2b4685#diff-0c4582f82c9ebb7e852f2d4999960c1e